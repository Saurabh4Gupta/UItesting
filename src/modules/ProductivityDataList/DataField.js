/* eslint-disable react/prop-types */
import React, { useState, useEffect } from 'react';
import { Box } from '@dentsu-ui/components';
import { useLocation } from 'react-router';
import PageController from '../PageController/PageController';
import DataList from './DataList';
import { dataFieldCms as PageContent } from '../../cms';
import { getData } from '../Mock/mockData';
import UploadFile from '../FileUpload/UploadFile';
import withPageController from '../../hoc/withPageController';

const DataField = (props) => {
  const location = useLocation();
  const state = { location }
  const { param  } = props;
  const query = new URLSearchParams(location.search);
  const clientCode = query.get('client_code');
  const [filterDataBy, setFilterDataBy] = useState({
    market: { label: 'All Markets', value: '' },
  });
  const [ongoingData, setDataList] = useState({ data: [], totalCount: 0 });
  const [isUploadModal, setIsUploadModal] = useState(false);
  const [isLoading, setLoading] = useState(true);
  const [originalOngingList, setOriginalOngoingList] = useState(
    getData(filterDataBy.market.value),
  );

  const updateOngoingList = (id, isToAdd) => {
    let filteredArray = [];

    const orignalArray = getData(filterDataBy.market.value).data;

    if (isToAdd) {
      const newArray = orignalArray.find((a) => a.id === id);

      filteredArray = [...originalOngingList.data];
      filteredArray.push(newArray);
    } else {
      filteredArray = originalOngingList.data.filter(
        (value) => value.id !== id,
      );
    }

    setOriginalOngoingList({
      data: filteredArray,
      totalCount: ongoingData.totalCount,
    });
  };

  if (state) {
    const requestId = { state }
    console.log('[REQUEST ID MOVED TO COMPLETE] =>', requestId)
  }


  const addNewRequest = (request) => {
    const orignalArray = getData(filterDataBy.market.value).data;
    const filteredArray = [...orignalArray, request];

    setOriginalOngoingList({
      data: filteredArray,
      totalCount: filteredArray.length,
    });
  };

  const handleMarket = (selected) => {
    setLoading(true);
    setFilterDataBy({ market: selected });
  };
  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
      setDataList(getData(filterDataBy.market.value));
      setOriginalOngoingList(getData(filterDataBy.market.value));
    }, 2000);
  }, [filterDataBy]);

  return (
    <>
      <PageController
        param={param}
        filterDataBy={filterDataBy}
        handleMarket={handleMarket}
        pageTitle=""
        pageMetadata="Client Overview"
        isCompleted=""
      >
        <Box mb="200px">
          <UploadFile
            cmsData={PageContent}
            modalOpen={isUploadModal}
            setModalOpen={setIsUploadModal}
          />
          <DataList
            cmsData={PageContent}
            market={filterDataBy.market}
            clientCode={clientCode}
            dataList={ongoingData}
            setDataList={setDataList}
            loading={isLoading}
            updateOngoingList={updateOngoingList}
            originalOngingList={originalOngingList}
            setOriginalOngoingList={setOriginalOngoingList}
            addNewRequest={addNewRequest}
          />
        </Box>
      </PageController>
    </>
  );
};

export default withPageController(DataField);
