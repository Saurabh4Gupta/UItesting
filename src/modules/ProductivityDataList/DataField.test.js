import React from 'react';
import { shallow } from 'enzyme';
import { DataField }  from './DataField';

describe('<EmptyState />', () => {
  let wrapper;
  const props = {
    param: [],
    useLocation: () => ({
        pathname: '/datafield',
        search: '?client_code=MC',
        hash: '',
        key: 't7nwlp',
    }),
  }
  beforeAll(() => {
    wrapper = shallow(<DataField {...props} />);
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
