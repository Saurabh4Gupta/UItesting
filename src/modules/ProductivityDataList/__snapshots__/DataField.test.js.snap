// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should match snapshot 1`] = `
<Fragment>
  <PageController
    clientCode="MC"
    history={Object {}}
    isToShowDataRequest={false}
    match={
      Object {
        "params": Object {
          "clientCode": "MC",
        },
        "path": "./url",
      }
    }
    setIsUploadModal={[Function]}
  />
  <Box
    m="45px"
    mb="200px"
  >
    <Memo(Overview) />
    <CreateData
      cmsData={
        Object {
          "actual": "actuals",
          "actualDataHint": "Select the number of months data required",
          "actualDataLabel": "Actual Productivity Data",
          "assignToHint": "Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.",
          "assignToLabel": "Assign To",
          "briefHint": "Tell us in detail what this data request is for, max 4000 characters",
          "briefLabel": "Briefing",
          "cancel": "Cancel",
          "complete": "Complete",
          "completeLabel": "Complete",
          "create": "Create",
          "createNewDataRequest": "Create new data request",
          "datePlaceHolder": "Date",
          "delete": "Delete",
          "dueDateHint": "Enter the date that you need this data request by",
          "dueDateLabel": "Due Date",
          "edit": "Edit",
          "editDataRequest": "Edit data request",
          "emptyCompletedProductivityDatarequestCaption": "No Productivity data requests have been moved to this section yet.",
          "emptyLabel": "Create new data request",
          "emptyProductivityDatarequestCaption": "There are currently no active Productivity data requests",
          "forecast": "forecast",
          "forecastDataHint": "Select the number of months data required",
          "forecastDataLabel": "Forecast Productivity Data",
          "labelFileTypeNotAllowed": "File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx",
          "lastUpdate": "Last Update",
          "localMarketHint": "Choose the market you wish to create this data request for",
          "localMarketLabel": "Local Market",
          "moveToComplete": "Move to Complete",
          "moveToOnGoing": "Move to Ongoing",
          "nameHint": "Make sure it's clear to help others identify it.Max 255 characters",
          "nameLabel": "Data request name",
          "ongoing": "Ongoing",
          "ongoingLabel": "Ongoing",
          "overdue": "Overdue",
          "productivityDatarequestCaption": " The ongoing and complete productivity data requests created for this client",
          "productivityDatarequestHeading": "Productivity data requests",
          "selectPlaceHolder": "Please Select",
          "templateFileHint": "Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less",
          "templateFileLabel": "Add tracker template file",
          "uploadButtonText": "Upload new file",
          "uploadFileLabel": "To upload a new file,drag and drop you file below or browse using the link",
          "uploadFileLargeMessage": "File must be less than 20MB",
          "viewClientsCaption": "Choose the client you wish to view",
          "viewClientsHeading": "My clients",
          "viewDetails": "View Details",
        }
      }
      handleModal={[Function]}
      isModalOpen={false}
      market=""
      setDataCreated={[Function]}
    />
    <UploadFile
      cmsData={
        Object {
          "actual": "actuals",
          "actualDataHint": "Select the number of months data required",
          "actualDataLabel": "Actual Productivity Data",
          "assignToHint": "Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.",
          "assignToLabel": "Assign To",
          "briefHint": "Tell us in detail what this data request is for, max 4000 characters",
          "briefLabel": "Briefing",
          "cancel": "Cancel",
          "complete": "Complete",
          "completeLabel": "Complete",
          "create": "Create",
          "createNewDataRequest": "Create new data request",
          "datePlaceHolder": "Date",
          "delete": "Delete",
          "dueDateHint": "Enter the date that you need this data request by",
          "dueDateLabel": "Due Date",
          "edit": "Edit",
          "editDataRequest": "Edit data request",
          "emptyCompletedProductivityDatarequestCaption": "No Productivity data requests have been moved to this section yet.",
          "emptyLabel": "Create new data request",
          "emptyProductivityDatarequestCaption": "There are currently no active Productivity data requests",
          "forecast": "forecast",
          "forecastDataHint": "Select the number of months data required",
          "forecastDataLabel": "Forecast Productivity Data",
          "labelFileTypeNotAllowed": "File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx",
          "lastUpdate": "Last Update",
          "localMarketHint": "Choose the market you wish to create this data request for",
          "localMarketLabel": "Local Market",
          "moveToComplete": "Move to Complete",
          "moveToOnGoing": "Move to Ongoing",
          "nameHint": "Make sure it's clear to help others identify it.Max 255 characters",
          "nameLabel": "Data request name",
          "ongoing": "Ongoing",
          "ongoingLabel": "Ongoing",
          "overdue": "Overdue",
          "productivityDatarequestCaption": " The ongoing and complete productivity data requests created for this client",
          "productivityDatarequestHeading": "Productivity data requests",
          "selectPlaceHolder": "Please Select",
          "templateFileHint": "Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less",
          "templateFileLabel": "Add tracker template file",
          "uploadButtonText": "Upload new file",
          "uploadFileLabel": "To upload a new file,drag and drop you file below or browse using the link",
          "uploadFileLargeMessage": "File must be less than 20MB",
          "viewClientsCaption": "Choose the client you wish to view",
          "viewClientsHeading": "My clients",
          "viewDetails": "View Details",
        }
      }
      modalOpen={false}
      setModalOpen={[Function]}
    />
    <DeleteData
      deleteRequest={[Function]}
      modalOpen={false}
      setModalOpen={[Function]}
    />
    <DataList
      clientCode="MC"
      cmsData={
        Object {
          "actual": "actuals",
          "actualDataHint": "Select the number of months data required",
          "actualDataLabel": "Actual Productivity Data",
          "assignToHint": "Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.",
          "assignToLabel": "Assign To",
          "briefHint": "Tell us in detail what this data request is for, max 4000 characters",
          "briefLabel": "Briefing",
          "cancel": "Cancel",
          "complete": "Complete",
          "completeLabel": "Complete",
          "create": "Create",
          "createNewDataRequest": "Create new data request",
          "datePlaceHolder": "Date",
          "delete": "Delete",
          "dueDateHint": "Enter the date that you need this data request by",
          "dueDateLabel": "Due Date",
          "edit": "Edit",
          "editDataRequest": "Edit data request",
          "emptyCompletedProductivityDatarequestCaption": "No Productivity data requests have been moved to this section yet.",
          "emptyLabel": "Create new data request",
          "emptyProductivityDatarequestCaption": "There are currently no active Productivity data requests",
          "forecast": "forecast",
          "forecastDataHint": "Select the number of months data required",
          "forecastDataLabel": "Forecast Productivity Data",
          "labelFileTypeNotAllowed": "File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx",
          "lastUpdate": "Last Update",
          "localMarketHint": "Choose the market you wish to create this data request for",
          "localMarketLabel": "Local Market",
          "moveToComplete": "Move to Complete",
          "moveToOnGoing": "Move to Ongoing",
          "nameHint": "Make sure it's clear to help others identify it.Max 255 characters",
          "nameLabel": "Data request name",
          "ongoing": "Ongoing",
          "ongoingLabel": "Ongoing",
          "overdue": "Overdue",
          "productivityDatarequestCaption": " The ongoing and complete productivity data requests created for this client",
          "productivityDatarequestHeading": "Productivity data requests",
          "selectPlaceHolder": "Please Select",
          "templateFileHint": "Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less",
          "templateFileLabel": "Add tracker template file",
          "uploadButtonText": "Upload new file",
          "uploadFileLabel": "To upload a new file,drag and drop you file below or browse using the link",
          "uploadFileLargeMessage": "File must be less than 20MB",
          "viewClientsCaption": "Choose the client you wish to view",
          "viewClientsHeading": "My clients",
          "viewDetails": "View Details",
        }
      }
      completeDataList={
        Object {
          "completedCount": 0,
          "completedData": Array [],
        }
      }
      handleDeleteModel={[Function]}
      handleModal={[Function]}
      handleTabIndex={[Function]}
      handleToggleData={[Function]}
      match={
        Object {
          "params": Object {
            "clientCode": "MC",
          },
          "path": "./url",
        }
      }
      ongoingDataList={
        Object {
          "data": Array [
            Object {
              "actualData": Object {
                "label": "4 months",
                "value": 4,
              },
              "assignTo": "UK Team",
              "briefing": "Need Data for the Q2 2020",
              "client": "Microsoft",
              "createdAt": "2020-08-14",
              "dueDate": "2020-08-25",
              "forecastData": Object {
                "label": "8 months",
                "value": 8,
              },
              "id": 1,
              "isActive": true,
              "isDeleted": false,
              "localMarket": Object {
                "label": "United Kingdom",
                "value": "UK",
              },
              "name": "Productivity Q2 2020",
              "quarter": "Q2",
              "status": "Overdue",
              "updatedAt": "29/10/20 at 14:32",
              "year": "2020",
            },
            Object {
              "actualData": Object {
                "label": "10 months",
                "value": 10,
              },
              "assignTo": "UK Team",
              "briefing": "Need Data for the Q3 2020",
              "client": "GM",
              "createdAt": "2020-08-14",
              "dueDate": "2020-12-30",
              "forecastData": Object {
                "label": "2 months",
                "value": 2,
              },
              "id": 2,
              "isActive": true,
              "isDeleted": false,
              "localMarket": Object {
                "label": "USA",
                "value": "USA",
              },
              "name": "Productivity Q3 2020",
              "quarter": "Q3",
              "status": "",
              "updatedAt": "30/10/20 at 14:32",
              "year": "2020",
            },
            Object {
              "actualData": Object {
                "label": "4 months",
                "value": 4,
              },
              "assignTo": "UK Team",
              "briefing": "Need Data for the Q2 2020",
              "client": "Microsoft",
              "createdAt": "2020-08-14",
              "dueDate": "2020-08-25",
              "forecastData": Object {
                "label": "8 months",
                "value": 8,
              },
              "id": 3,
              "isActive": true,
              "isDeleted": false,
              "localMarket": Object {
                "label": "United Kingdom",
                "value": "UK",
              },
              "name": "Productivity Q2 2020",
              "quarter": "Q2",
              "status": "Overdue",
              "updatedAt": "29/10/20 at 14:32",
              "year": "2020",
            },
          ],
          "totalCount": 3,
        }
      }
      tabIndex={0}
    />
  </Box>
</Fragment>
`;
